config-model/src/main/java/com/yahoo/vespa/model/container/ContainerCluster.java

```diff
@@ -304,6 +307,38 @@ public void addJaxProviders() {
         addSimpleComponent(XPathFactoryProvider.class);
     }
 
+    public void addIdentity() {
+        if (getDeploymentSpec().isPresent()) {
+            DeploymentSpec deploymentSpec = getDeploymentSpec().get();
+            if (deploymentSpec.athenzDomain().isPresent()) {
+                String domain = deploymentSpec.athenzDomain().get();
+                // Service can be configured globally, or per zone
+                String service = deploymentSpec.zones().stream()
+                        .filter(zone -> zone.deploysTo(getZone().environment(), Optional.of(getZone().region())))
+                        .findAny()
+                        .flatMap(DeploymentSpec.DeclaredZone::athenzService)
+                        .orElse(deploymentSpec.athenzService().orElseThrow(() -> new RuntimeException("Missing Athenz service configuration")));
+
+                // Set lbaddress, or use first hostname if not specified.
+                HostName lbName = Optional.ofNullable(getRoot().getDeployState().getProperties().loadBalancerName())
+                        .orElseGet(
+                                () -> HostName.from(getRoot().getDeployState().getProperties().configServerSpecs().stream()
+                                                            .findFirst()
+                                                            .map(ConfigServerSpec::getHostName)
+                                                            .orElse("unknown") // Currently unable to test this, hence the unknown
+                                ));
+
+                Identity identity = new Identity(domain.trim(), service.trim(), lbName);
```

Comment:
If 'Identity' is a representation of the 'IdentityProvider' component, I think we should rename the config model class as well.